app/src/main/java/com/checkin/app/checkin/Home/NearbyRestaurantAdapter.kt

```diff
@@ -1,43 +1,155 @@
         holder.bindData(data[position])
     }
 
     inner class NearbyRestaurantViewHolder(itemView: View) : BaseViewHolder<NearbyRestaurantModel>(itemView) {
+
+
+        var imageView:ImageView=itemView.findViewById(R.id.iv_home_restaurant_banner_main)
+
+        var image_title:TextView=itemView.findViewById(R.id.tv_home_restaurant_banner_image_title)
+
+        var ratings:TextView=itemView.findViewById(R.id.tv_home_restaurant_banner_ratings)
+        var resturantName:TextView=itemView.findViewById(R.id.tv_home_restaurant_banner_restaurant_name)
+
+        init {
+
+
+            imageView.scaleType = ImageView.ScaleType.FIT_XY
+        }
+
         override fun bindData(data: NearbyRestaurantModel) {
-            TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+
+            image_title.setText(data.imageTitle)
+            ratings.setText(data.ratings)
+            resturantName.setText(data.restaurantName)
+            Glide.with(context).asBitmap()
+                    .load(
+                    data.url).into(imageView)
+
+
         }
     }
 
     inner class AdvertisementViewHolder(itemView: View) : BaseViewHolder<Any>(itemView) {
+        var vpBanner: ViewPager=itemView.findViewById(R.id.vp_home_banner)
+       internal val mPagerAdapter: BannerPagerAdapter=BannerPagerAdapter(context);
+        val indicatorView: PageIndicatorView? = itemView.findViewById(R.id.indicator_home_banner)
+        init {
+
+
+            vpBanner.adapter = mPagerAdapter
+            if (indicatorView != null) {
+                indicatorView.setViewPager(vpBanner)
+
+                indicatorView.setAnimationType(AnimationType.FILL)
+                indicatorView.setClickListener { position -> vpBanner.currentItem = position }
+            } else {
+
+            }
+            vpBanner.addOnPageChangeListener(object : ViewPager.OnPageChangeListener {
+                override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {}
+
+                override fun onPageSelected(position: Int) {}
+
+                override fun onPageScrollStateChanged(state: Int) {
+                    enableDisableSwipeRefresh(state == ViewPager.SCROLL_STATE_IDLE)
+                }
+            })
+        }
+
         override fun bindData(data: Any) {
-            TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+
+
+
+
+
         }
+        fun enableDisableSwipeRefresh(enable: Boolean) {  
+        }
+
     }
+    internal inner class BannerPagerAdapter(var mContext: Context) : PagerAdapter() {
+        var mResources = intArrayOf(R.drawable.first_banner, R.drawable.second_banner, R.drawable.third_banner)
+        var mLayoutInflater: LayoutInflater
+
+        init {
+            mLayoutInflater = mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
```

Comment:
Why did you use this?

Reply:
That code is not recommended to use anymore... `LayoutInflator.from` is used always